hw/ip/flash_ctrl/doc/_index.md

```diff
@@ -65,40 +75,130 @@ The physical controller supports the following features
 *  Data and informational paritions within each bank of flash memory
 *  Arbitration between host requests and controller requests at the bank level
    *  Host requests are always favored, however the controller priority can escalate if it repeatedly loses arbitration
-   *  Since banks are arbitrated independently, where transactions may take different amounts of times to complete, the physical controller is also responsible for ensuring in-order response to both the controller and host.
+   *  Since banks are arbitrated independently and transactions may take different amounts of times to complete, the physical controller is also responsible for ensuring in-order response to both the controller and host.
 *  Flash read stage
-   *  Each bank maintains a parameterizable number of read buffers in front of the flash memory
+   *  Each bank maintains a parameterizable number of read buffers in front of the flash memory (default to 4)
    *  The read buffers behave as miniature read-only-caches to store flash data when flash words are greater than bus words.
+   *  When a program or erase collides with an entry stored in the read buffer, the buffer contents are invalidated.
```

Comment:
Actually, I am not entirely sure I understand this statement. I thought at a time, you can only have one operation running from the controller. By read buffer, do you mean the buffer on the host interface (which I believe is different from the read buffer on the controller side)? Collision can only occur from the host interface.

Reply:
i should probably clarify this a little.  

The read buffer is inside the `flash_phy` and its the basic "read-only" cache. Since it's a multi-entry cache (4), it's possible for the contents stored there to collide with an ongoing program or erase. So collisions can occur from both the host and the controller.

The read "buffer" inside `flash_ctrl` is really just a FIFO, and it has no concept of race conditions, and i don't think it needs to be because it is one at a time. 

If you preview the doc and look at the block diagram, you'll see the buffers in `flash_phy` i'm referring to.

Reply:
added more text to explain.